site/_posts/2020-12-21-fusionauth-lets-iot-firm-focus-their-app.md

```diff
@@ -0,0 +1,102 @@
+
+**Dan:** Would you be interested in open sourcing the tooling around email messages localization :) ? That sounds pretty great.
+
+**Michał:** I'm not entirely sure whether those few lines in python are worth publishing, but I can share what they are about. The idea is to create jinja2 templates for each email. The templates are localized with the use of GNU gettext. Based on the jinja2 templates, localized FreeMarker templates are rendered. So, yeah - templates for templates ;) 
+
+Rendered templates and stylesheets are then passed through the premailer tool to increase visual consistency across email clients and pushed into FusionAuth either via Kickstart or the API. If you would be interested in building such functionality I believe it should be done in a similar way that themes are localized. That is, each message has an HTML and Text template and messages file for each locale. Our current solution is just a workaround of the current limitation and as such is not particularly elegant.
+
+> In order not to get distracted while building something awesome, we didn't want to implement front-end aspects of the authorization flow, which can be pretty complex taking different authentication flows into account. 
+
+**Dan:** Which reverse proxy did you use?
+
+**Michał:** HAProxy as an edge proxy with TLS offloading, and a bunch of nginx instances for static content, WSGI applications and the FusionAuth instance itself. HAProxy also has some neat features in the terms of upstream service health checking. 
+
+The results of those checks are easily exposed in prometheus format. Since FusionAuth doesn't expose such metrics itself, we rely on the data provided by HAproxy. Anyway, a prometheus interface for FusionAuth would be lovely! [ed note: follow along on this issue for [prometheus support](https://github.com/FusionAuth/fusionauth-issues/issues/362).]
+
+**Dan:** Why did you choose FusionAuth over the competition?
+
+**Michał:** We didn't want to go with a SaaS option, so we were looking for a self-hosted solution. We defined a set of requirements including ease of user migration, MFA options, and so on. The features we needed were, in general, present in the solutions available on the market, but rarely all at once. FusionAuth is a product where each of our requirements is either fulfilled out of the box, or it could be easily developed based on the API, webhooks or other workarounds.
+
+Most of the concerns about FusionAuth stemmed from the closed source model. Apart from issues that may occur if FusionAuth goes out of business, the closed source model also closes the way to perform quick fixes in the service if glitches are encountered. After some time as a member of the FusionAuth community I've seen that the dev team is pretty responsive to users' feedback, so it calmed me enough to be confident in choosing FusionAuth as a central identity management service for our product.
```

Comment:
Most of the concerns about FusionAuth stemmed from the closed source model. Apart from issues that may occur if FusionAuth goes out of business, the closed source model also closes the way to perform quick fixes in the service if glitches are encountered. --> Most of the concerns about FusionAuth stemmed from the closed source model such as the way to perform quick fixes in the service if glitches are encountered.

Reply:
Changed to: "Most of the concerns about FusionAuth stemmed from the closed source model, particularly around responsiveness to bugs and business model risk."